mlir/lib/Conversion/SCFToEmitC/SCFToEmitC.cpp

```diff
@@ -322,7 +322,11 @@ void SCFToEmitCPass::runOnOperation() {
   // Fallback converter
   // See note https://mlir.llvm.org/docs/DialectConversion/#type-converter
   // Type converters are called most to least recently inserted
-  typeConverter.addConversion([](Type t) { return t; });
+  typeConverter.addConversion([](Type type) -> std::optional<Type> {
+    if (emitc::isSupportedEmitCType(type))
+      return type;
+    return {};
```

Comment:
```suggestion
    if (!emitc::isSupportedEmitCType(type))
      return {};
    return type;
```
Might want to inverse the check to allow additional checks if needed later?

With this change we might also want to update the comment in L322-L324.

Reply:
What the comments change to?

Reply:
Fine to what you updated it. Thanks.